lib/spree_i18n/ransack_translator.rb

```diff
@@ -0,0 +1,64 @@
+module SpreeI18n
+  class RansackTranslator
+    def initialize(klass, params)
+      @klass = klass
+      @params = params
+    end
+
+    def translated_params
+      @params.each_with_object({}) do |(key, value), result|
+        names, pred = split_key(key)
+        translated_names = translate_names(names)
+        translated_key = join_key translated_names, pred
+        result[translated_key] = value
+      end
+    end
+
+    private
+
+    def split_key(key)
+      stripped_name = key.to_s.dup
+      pred = Ransack::Predicate.detect_and_strip_from_string!(stripped_name)
+
+      names = stripped_name.split(/_or_/)
+      [names, pred]
+    end
+
+    def join_key(names, pred)
+      key = names.join('_or_')
+      key += "_#{pred}" unless pred == ''
+      key
+    end
+
+    def translate_names(names)
+      names.map { |name| translated_name(name) }
+    end
+
+    def translated_name(name)
+      model, attrib_name = locate_attribute name
+      prefix = name.chomp(attrib_name)
+
+      if (model.try(:translated_attribute_names) || []).include? attrib_name.to_sym
```

Comment:
I _think_ changing this line to this could fix the errors:

```ruby
      if model && model.translated_attribute_names.include?(attrib_name.to_s.to_sym)
```

I can't say I get exactly whats going on with this `locate_attribute`, I suppose there's no other check we could add to avoid hitting this stack when we dont have too right? seems a bit expensive and very complicated